examples/text_labels_and_annotations/custom_legends.py

```diff
@@ -0,0 +1,56 @@
+"""
+========================
+Composing Custom Legends
+========================
+
+Composing custom legends piece-by-piece.
+
+Sometimes you don't want a legend that is explicitly tied to data that
+you have plotted. For example, say you have plotted 10 lines, but don't
+want a legend item to show up for each one. If you simply plot the lines
+and call ``ax.legend()``, you will get the following:
+"""
+# sphinx_gallery_thumbnail_number 2
+from matplotlib import rcParams, cycler
+import matplotlib.pyplot as plt
+import numpy as np
+
+N = 10
+data = [np.logspace(0, 1, 100) + np.random.randn(100) + ii for ii in range(N)]
+data = np.array(data).T
+cmap = plt.cm.coolwarm
+rcParams['axes.prop_cycle'] = cycler(color=cmap(np.linspace(0, 1, N)))
+
+fig, ax = plt.subplots()
+lines = ax.plot(data)
+ax.legend(lines, loc=(1.05, .05), )
+
+##############################################################################
+# In this case, we can compose a legend using Matplotlib objects that aren't
+# explicitly tied to the data that was plotted. For example:
+
+from matplotlib.lines import Line2D
+custom_lines = [Line2D([0], [0], color='b', lw=4),
+                Line2D([0], [0], color='grey', lw=4),
+                Line2D([0], [0], color='r', lw=4)]
+
+fig, ax = plt.subplots()
+lines = ax.plot(data)
+ax.legend(custom_lines, ['Cold', 'Medium', 'Hot'], loc=(1.05, .8),)
+
+
+###############################################################################
+# There are many other Matplotlib objects that can be used in this way. Below
+# is a list of common ones.
+
+from matplotlib.patches import Patch
+from matplotlib.lines import Line2D
```

Comment:
Maybe a dumb comment, but Line2D has already been loaded, hasn't it?

Reply:
yeah here I was loading it again just so that it's easier for people to trace the imports in case they wanna do it themselves. Think it should be removed? I agree it's not standard PEP style

Reply:
Well I do not know what exactly are our guidelines about this kind of things. I am not particularly pushing for removing it, I just wanted to be sure it had been done on purpose.

About PEP8, there are other imports that are not done at the top of the file anyway, so I guess that it does not really matter anyway ^^.

=> my call would be to keep it this way except if somebody more involved in the docs than I am complains ;)

Reply:
I think @choldgraf might be the one _most_ involved in the docs. :P

Reply:
@QuLogic I know ^^, but some other core devs are also quite involved about the docs and do care about what are the relevant guidelines that the docs should follow (hint: from my perspective, you are counted in that group :P)